server/test/connected_objects.go

```diff
@@ -0,0 +1,668 @@
+											{Userset: &openfgapb.Userset_TupleToUserset{
+												TupleToUserset: &openfgapb.TupleToUserset{
+													Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+													ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+												},
+											}},
+										},
+									},
+								},
+							},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "folder"},
+									},
+								},
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:folder2", "parent", "folder:folder1"),
+				tuple.NewTupleKey("folder:folder3", "parent", "folder:folder2"),
+			},
+			expectedObjects: []string{"folder:folder1", "folder:folder2", "folder:folder3"},
+		},
+		{
+			name: "Test-9",
+			request: &commands.ConnectedObjectsRequest{
+				StoreID:    id.Must(id.New()).String(),
+				ObjectType: "folder",
+				Relation:   "viewer",
+				User:       &openfgapb.ObjectRelation{Object: "user:jon"},
+			},
+			resolveNodeLimit: 2,
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "folder",
+						Relations: map[string]*openfgapb.Userset{
+							"parent": {Userset: &openfgapb.Userset_This{}},
+							"viewer": {
+								Userset: &openfgapb.Userset_Union{
+									Union: &openfgapb.Usersets{
+										Child: []*openfgapb.Userset{
+											{Userset: &openfgapb.Userset_This{}},
+											{Userset: &openfgapb.Userset_TupleToUserset{
+												TupleToUserset: &openfgapb.TupleToUserset{
+													Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+													ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+												},
+											}},
+										},
+									},
+								},
+							},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "folder"},
+									},
+								},
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:folder2", "parent", "folder:folder1"),
+				tuple.NewTupleKey("folder:folder3", "parent", "folder:folder2"),
+			},
+			expectedError: serverErrors.AuthorizationModelResolutionTooComplex,
```

Comment:
I don't understand why this would fail. `folder1` is parent of `folder2` which is parent of `folder3`. `jon` can view folder1. Therefore, `jon` can also view folder2 and folder3?

Reply:
Because of the `resolveNodeLimit`. The resolution depth is constrained to a depth of 2, but the nested hierarchy contains a depth of 3, and so it returns an error.

Reply:
Renamed the test for clarity on this one.